src/Extensions/StagingExtension.php

```diff
@@ -15,16 +14,9 @@ class StagingExtension extends SiteTreeExtension
     public function contentcontrollerInit(ContentController $controller): void
     {
         // If we are currently browsing in a CMSPreview, then we do not want to write or publish any CacheKeys
-        if ($controller->getRequest()->getVar('CMSPreview')) {
+        if ($controller->getRequest()->getVar('CMSPreview')) { // phpcs:ignore
             StagingState::singleton()->disableWrite();
             StagingState::singleton()->disablePublish();
-
-            return;
-        }
-
-        // If we are browsing in stage=Stage, then we do not want to publish any CacheKeys
-        if ($controller->getRequest()->getVar('stage') === Versioned::DRAFT) { // phpcs:ignore
-            StagingState::singleton()->disablePublish();
```

Comment:
Moved logic to `StagingState::canPublish()` so that this behaviour is global and not only when we hit `contentcontrollerInit()`.